src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx

```diff
@@ -2,3 +2,572 @@ export const meta = {
+    priority: {
+      eq: 1
+    }
+  }
+};
+
+await API.graphql({ 
+  query: listTodos, variables: variables 
+});;
+```
+
+</Block>
+</BlockSwitcher>
+
+#### Compound filters
+
+You can combine filters with `and`, `or`, and `not` Boolean logic. Observe, in the autogenerated schema above, that `ModelTodoFilterInput` is recursive in respect to those fields. So if, for example, you wanted to filter for `priority` values of 1 OR 2, you would do this:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { ListTodosQueryVariables, ListTodosQuery } from './API';
+
+const variables: ListTodosQueryVariables = {
+  filter: {
+    or: [
+      { priority: { eq:1 } },
+      { priority: { eq:2 } }
+    ]
+  }
+};
+
+await API.graphql<GraphQLQuery<ListTodosQuery>>({ 
+  query: listTodos, variables: variables 
+});
+```
+
+</Block>
+
+<Block name="JavaScript">
+
+```js
+const variables = {
+  filter: {
+    or: [
+      { priority: { eq:1 } },
+      { priority: { eq:2 } }
+    ]
+  }
+};
+
+await API.graphql({ 
+  query: listTodos, variables: variables 
+});;
+```
+
+</Block>
+</BlockSwitcher>
+
+Note that querying for `priority` of 1 and 2 would return no results, because this is Boolean logic instead of natural language.
+
+### Paginating queries
+
+To paginate results in AWS AppSync, make a query request specifying the `nextToken` field in the selection set and by passing in a `limit` input variable. The `limit` input variable limits how many results are returned. The response will include a `nextToken` you can use to request the next page of data. A `nextToken` is a very long string that represents the cursor to the starting item of the next query made with these filters.
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { ListTodosQueryVariables, ListTodosQuery } from './API';
+
+// Fetch first 20 records
+const variables: ListTodosQueryVariables = {
+  limit: 20, 
+  // add filter as needed
+};
+
+const res = await API.graphql<GraphQLQuery<ListTodosQuery>({ 
+  query: listTodos, variables: variables
+});
+
+const { items: itemsPage1, nextToken } = res.data?.listTodos;
+
+// Fetch the next 20 records
+variables.nextToken = nextToken;
+
+const res = await API.graphql<GraphQLQuery<ListTodosQuery>({ 
+  query: listTodos, variables: variables 
+});
+
+const { items: itemsPage2 } = res.data?.listTodos;
+```
+
+</Block>
+
+<Block name="JavaScript">
+
+```js
+// Fetch first 20 records
+const variables = {
+  limit: 20, 
+  // add filter as needed
+};
+
+const res = await API.graphql({ 
+  query: listTodos, variables: variables
+});
+
+const { items: itemsPage1, nextToken } = res.data.listTodos;
+
+// Fetch the next 20 records
+variables.nextToken = nextToken;
+
+const res = await API.graphql({ 
+  query: listTodos, variables: variables
+});
+
+const { items: itemsPage2 } = res.data.listTodos;
+```
+
+</Block>
+</BlockSwitcher>
+
+<Accordion title='Implement pagination in your GraphQL API' headingLevel='3' eyebrow='Walkthrough'>
```

Comment:
@heatheramz - it's worth considering to remove this accordion. Rereading it, I feel like it's just a more expanded version of the code snippet above without additional value/detail.

Reply:
@renebrandel  We can - this is from the guide - https://docs.amplify.aws/guides/api-graphql/graphql-pagination/q/platform/js/ which is an area we are looking to merge content or deprecate as needed. If the updated guidance here provides the same then the recommendation will be to retire the JS guide (and other platforms as this page is expanded).  Is there anything from the accordion that we would want to add into the main page before it is deleted?

Reply:
Discuss 9/26 - keeping for now - add to UX if possible